packages/wonder-blocks-birthday-picker/src/components/__tests__/birthday-picker.test.js

```diff
@@ -0,0 +1,343 @@
+                name: "Aug",
+            });
+            userEvent.click(monthOptionNew, undefined, {
+                skipPointerEventsCheck: true,
+            });
+
+            userEvent.click(screen.getByTestId("birthday-picker-day"));
+            const dayOptionNew = await screen.findByRole("option", {name: "9"});
+            userEvent.click(dayOptionNew, undefined, {
+                skipPointerEventsCheck: true,
+            });
+
+            userEvent.click(screen.getByTestId("birthday-picker-year"));
+            const yearOptionNew = await screen.findByRole("option", {
+                name: "2020",
+            });
+            userEvent.click(yearOptionNew, undefined, {
+                skipPointerEventsCheck: true,
+            });
+
+            // Assert
+            expect(onChange).toHaveBeenLastCalledWith("2020-08-09");
+        });
+
+        it("onChange triggers when a new value is selected after a default value is set", async () => {
+            // Arrange
+            const onChange = jest.fn();
+
+            // Act
+            render(
+                <BirthdayPicker
+                    defaultValue="2017-07-17"
+                    onChange={onChange}
+                />,
+            );
+
+            userEvent.click(screen.getByTestId("birthday-picker-month"));
+            const monthOption = await screen.findByRole("option", {
+                name: "Aug",
+            });
+            userEvent.click(monthOption, undefined, {
+                skipPointerEventsCheck: true,
+            });
+
+            userEvent.click(screen.getByTestId("birthday-picker-day"));
+            const dayOption = await screen.findByRole("option", {name: "9"});
+
+            userEvent.click(dayOption, undefined, {
+                skipPointerEventsCheck: true,
+            });
+
+            userEvent.click(screen.getByTestId("birthday-picker-year"));
+            const yearOption = await screen.findByRole("option", {
+                name: "2018",
+            });
+            userEvent.click(yearOption, undefined, {
+                skipPointerEventsCheck: true,
+            });
+
+            // Assert
+            expect(onChange).toHaveBeenCalledWith("2018-08-09");
+        });
+
+        it("onChange triggers with null when an invalid value is selected after a default value is set", () => {
+            // Arrange
+            const onChange = jest.fn();
+            let maybeInstance: ?BirthdayPicker = null;
+
+            // Act
+            render(
+                <BirthdayPicker
+                    defaultValue="2017-07-17"
+                    onChange={onChange}
+                    ref={(node) => (maybeInstance = node)}
+                />,
+            );
+
+            if (!maybeInstance) {
+                throw new Error("BirthdayPicker instance is undefined");
+            }
+            const instance = maybeInstance;
+
+            // This test was written by calling methods on the instance because
```

Comment:
Useful comment, thanks!

Reply:
oh oops, that's was John's comment, but yeah, it's supper helpful :)

I forgot to mention the minor detail that I moved these tests from webapp to wb.